check-licenses/action.yml

```diff
@@ -0,0 +1,28 @@
+name: "Check licenses"
+description: "Check if package licenses in the current environment are among the accepted ones"
+
+inputs:
+  accepted-licenses:
+    description: "List of accepted licenses"
+    default: "MIT BSD Apache PSF HPND"
+    required: false
+    type: string
```

Comment:
What about ignored packages? Shouldn't we provide them as a list as well? The actions should also have a record file of accepted packages whose licenses are accepted (even if they are not of the types accepted by default). Don't know if you follow me but we can talk about this offline =)

Reply:
yep, I'm on it. So far I'm only excluding matplotlib for testing purposes.

Now I'm working on making it fail if bad licenses are found. After that I will stop hard-coding the inputs and pass them properly. I am planning on adding these inputs:
- ``accepted-licenses``: with some default well-known licenses (I took the ones you had [here](https://github.com/pyansys/pyansys/blob/main/.github/workflows/dependency-reviewer.yml#L22), @RobPasMue).
- ``additional-accepted-licenses``: just in case some projects want to add one or two more to the ``accepted-licenses`` list.  Instead of having to re-write the entire list.
- ``ingored-packages``: list of packages that are trusted but have invalid licenses.
-  ``python-version``
- ``skip-install``: to avoid installing the package if this action is used inside another one that already installs the package

Then I will add the install package step before checking the licenses.

Do you think I'm missing something? Comments?

Reply:
Well... It's confusing for me to have two arguments called ``accepted-licenses`` and ``additional-accepted-licenses``.

* The first ones (``accepted-licenses``) should be hard-coded in the action. In fact we should be reading them from a file (same way you execute a ``check-licenses.py`` file) so that it is more maintainable. We should keep track of them in a txt file or similar of the accepted licences by the PyAnsys repositories.
* The ``additional-accepted-licenses`` approach is good for those repos which have specific needs.
* The ``ignored-packages`` option should behave similarly to the previous two. We should have a list of generally accepted packages that have "wrong" licenses. And we should also allow users to provide additional packages (if needed)

Keep in mind that this action is oriented for PyAnsys repositories. There is no problem on tailoring it for our purpose - with these ``accepted-licenses.txt`` and ``ignored-packages.txt`` files, for example.

Reply:
Don't know if you got my point but basically I am proposing to:
* Delete ``accepted-licenses`` and ``ignored-packages`` options
* Only accept ``additional-accepted-licenses`` and ``additional-ignored-packages``
* Write to a TXT file those licenses accepted by default, and read them from there in the action
* Write to a TXT file those packages accepted by default, and read them from there in the action

Reply:
After an off-line conversation, we've decided to go for the txt approach.

Thus, the inputs of this actions are doing to be:
- ``python-version``
- ``skip-install``: to avoid installing the package if this action is used inside another one that already installs the package

Inside the action we're going to hard-code the most common licenses. Then, two txt files (``accepted-licenses.txt`` and ``ignored-packages.txt``) will be used in order to tweak which extra licenses are also accepted and which packages can be ignored, depending on the particularities of each repo.

Reply:
pinging @jorgepiloto and @MaxJPRey for further input on the final decision.

Reply:
My main point is that the default accepted licenses and ignored packages should already be defined inside this action (if possible, as TXT files). The way in which users provide additional licenses and ignored packages is up for discussion, either with an additional TXT file which they maintain in their own repos or directly in the YML file calling this action.